### PR TITLE
Moving automation into workbench libs

### PR DESCRIPTION
**Pull request by UrsaStutsman**
\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: automation/docker/application.conf.ctmpl [hunk 1]
```diff
@@ -3,6 +3,7 @@
 {{with $environment := env "ENVIRONMENT"}}
 {{with $dir := env "ROOT_DIR"}}
 {{with $local_ui := env "LOCAL_UI"}}
+{{with $trialBillingPemSecrets := vault (printf "secret/dsde/firecloud/%s/common/trial-billing-account.pem" $environment)}}
```
Comment (jmthibault79) [inline automation/docker/application.conf.ctmpl]: Most of these additions are not used by Leonardo automation, but they are required by the current iteration of workbench-libs automation config.  Removing these is a TODO but let's not block on it.

File: automation/project/Dependencies.scala [hunk 2]
```diff
@@ -35,13 +37,14 @@ object Dependencies {
     "com.typesafe.akka"   %%  "akka-testkit"        % akkaV     % "test",
     "com.typesafe.akka"   %%  "akka-slf4j"          % akkaV,
     "org.specs2"          %%  "specs2-core"   % "3.7"  % "test",
-    "org.scalatest"       %%  "scalatest"     % "2.2.6"   % "test",
+    "org.scalatest"       %%  "scalatest"     % "3.0.1"   % "test",
```
Comment (jmthibault79) [inline automation/project/Dependencies.scala]: This update to match `workbench-service-test` required changes in how we use `eventually`

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/FireCloudView.scala [hunk 3]
```diff
@@ -1,6 +1,6 @@
 package org.broadinstitute.dsde.firecloud.page
 
-import org.broadinstitute.dsde.workbench.util.WebBrowserUtil
```
Comment (jmthibault79) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/FireCloudView.scala]: TODO non-blocker: Remove most of the `page` classes from Leonardo

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala [hunk 4]
```diff
@@ -119,9 +124,10 @@ trait LeonardoTestUtils extends WebBrowserSpec with Matchers with Eventually wit
     clusterCheck(Leonardo.cluster.get(googleProject, clusterName), clusterRequest.labels, googleProject, clusterName, Seq(ClusterStatus.Creating), clusterRequest.jupyterExtensionUri)
 
     // wait for "Running" or error (fail fast)
+    implicit val patienceConfig: PatienceConfig = clusterPatience
     val actualCluster = eventually {
       clusterCheck(Leonardo.cluster.get(googleProject, clusterName), clusterRequest.labels, googleProject, clusterName, Seq(ClusterStatus.Running, ClusterStatus.Error), clusterRequest.jupyterExtensionUri)
-    } (clusterPatience)
```
Comment (jmthibault79) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: ScalaTest 3 makes it more difficult to pass `patience` as a parameter

File: docker/build.sh [hunk 5]
```diff
@@ -37,20 +37,30 @@ function make_jar()
 function docker_cmd()
 {
     if [ $DOCKER_CMD = "build" ] || [ $DOCKER_CMD = "push" ]; then
-        echo "building leonardo docker image..."
-        GIT_SHA=$(git rev-parse ${GIT_BRANCH})
-        echo GIT_SHA=$GIT_SHA > env.properties  # for jenkins jobs
-        docker build -t $REPO:${GIT_SHA:0:12} .
+        echo "building $PROJECT docker image..."
```
Comment (jmthibault79) [inline docker/build.sh]: These changes enable new-style Jenkins builds

**Comment by jmthibault79** [comment]
jenkins retest

**Comment by jmthibault79** [comment]
Initial Jenkins-pipeline failed because clusters were in Error state.  Trying again.

**Comment by jmthibault79** [comment]
jenkins retest

File: automation/docker/application.conf.ctmpl (hunk 1 above)
Comment (helgridly) [reply automation/docker/application.conf.ctmpl]: should vault-y confs be living in a public repo?

File: automation/project/Dependencies.scala [hunk 6]
```diff
@@ -7,11 +7,13 @@ object Dependencies {
 
   val workbenchModelV   = "0.8-d97f551"
   val workbenchGoogleV  = "0.10-b95b2c1"
+  val serviceTestV = "0.1-c5a3162-SNAP"
```
Comment (rtitle) [inline automation/project/Dependencies.scala]: Should update to non-SNAP; I guess that is dependent on the workbench-libs PR being merged.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/FireCloudView.scala (hunk 3 above)
Comment (helgridly) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/FireCloudView.scala]: can we github ticket these todos?

File: automation/project/Dependencies.scala [hunk 7]
```diff
@@ -7,11 +7,13 @@ object Dependencies {
...
 
   val excludeWorkbenchModel =   ExclusionRule(organization = "org.broadinstitute.dsde.workbench", name = "workbench-model_2.11")
 
   val workbenchModel: ModuleID =  "org.broadinstitute.dsde.workbench" %% "workbench-model"  % workbenchModelV
   val workbenchGoogle: ModuleID = "org.broadinstitute.dsde.workbench" %% "workbench-google" % workbenchGoogleV excludeAll excludeWorkbenchModel
+  val workbenchServiceTest: ModuleID = "org.broadinstitute.dsde.workbench" %% "workbench-service-test" % serviceTestV % "test" classifier "tests" excludeAll excludeWorkbenchModel
```
Comment (rtitle) [inline automation/project/Dependencies.scala]: Wonder if we should be excluding both `workbenchModel` and `workbenchGoogle`?

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/config/LeoAuthToken.scala [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport
+import com.google.api.client.json.jackson2.JacksonFactory
+import org.broadinstitute.dsde.workbench.auth.AuthToken
+
+import scala.collection.JavaConverters._
+
+case class LeoAuthToken(value: String)
+
+case object LeoAuthToken extends AuthToken {
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/config/LeoAuthToken.scala]: I am curious how this class is different from `AuthToken`, `UserAuthToken`, `ServiceAccountAuthToken` classes in workbench-libs:

https://github.com/broadinstitute/workbench-libs/tree/2801-sam/serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/auth

Could one of those classes be used instead?

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala [hunk 9]
```diff
@@ -28,13 +29,17 @@ trait LeonardoTestUtils extends WebBrowserSpec with Matchers with Eventually wit
   val swatTestBucket = "gs://leonardo-swat-test-bucket-do-not-delete"
   val incorrectJupyterExtensionUri = swatTestBucket + "/"
 
-  // Ron and Hermione are on the dev Leo whitelist.
-  val ronAuthToken = AuthToken(LeonardoConfig.Users.ron)
-  val hermioneAuthToken = AuthToken(LeonardoConfig.Users.hermione)
-  val ronEmail = LeonardoConfig.Users.ron.email
+  // Ron and Hermione are on the dev Leo whitelist, and Hermione is a Project Owner
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: What whitelist? :)

Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: Heh, good point.  I would like to defer those changes please :)

Comment (rtitle) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: +1

File: automation/docker/application.conf.ctmpl (hunk 1 above)
Comment (jmthibault79) [reply automation/docker/application.conf.ctmpl]: every repo will be getting this

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/FireCloudView.scala (hunk 3 above)
Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/FireCloudView.scala]: yep I'll write something up

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/config/LeoAuthToken.scala (hunk 8 above)
Comment (UrsaStutsman) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/config/LeoAuthToken.scala]: We've got a card already for the iteration of this - I can add refactoring this to it

Comment (UrsaStutsman) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/config/LeoAuthToken.scala]: And I've added it.

**Pull request merged**